hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -644,6 +655,9 @@ private OzoneManager(OzoneConfiguration conf, StartupOption startupOption)
       kmsProvider = null;
       LOG.error("Fail to create Key Provider");
     }
+    if (isLeaderReady() && kmsProvider != null) {
```

Comment:
This will only work if the node is already a leader.
what if a follower transition into leader? Will that scenario be covered?

Reply:
> what if a follower transition into leader? Will that scenario be covered?

No, during restart it will see which is leader and initializeEdekCache(conf) will be executed on that node!

Reply:
But that's after a restart. A follower can transition into leader while it's still up.
Is there a callback from Ratis when a role becomes a leader? There are certain things we want leader to take on as soon as it becomes a leader.

Reply:
So, do we want the cache to warm up upon OM restart or when the leader changes?

> There are certain things we want leader to take on as soon as it becomes a leader.

What operations we want to take care of as soon as OM becomes a leader?

Reply:
I have made a change to warm up the EDEK cache on OM leader change.